features/audio-playback/index.html

```diff
@@ -0,0 +1,171 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width">
+  <title>Audio playback</title>
+</head>
+<body>
+<p><a href="../index.html">[Home]</a></p>
+
+  <p>For testing fingerprinting, ensure that both sound the same and playing the copy several times also sounds the same but makes a different hash.</p>
+
+  <main>
+    <audio src="sample.mpga" id="sample"></audio>
+    <button id="first">Play original</button>
+    <button id="second">Playback copy</button>
+    <div id="stats">
+    </div>
+  </main>
+
+  <style>
+      #canvas, #output {
+          width: 300px;
+          height: 300px;
+          border: 1px solid black;
+      }
+      #canvas {
+          border-color: red;
+      }
+      #stats {
+          white-space: pre;
+      }
+      textarea {
+          width: 600px;
+          height: 600px;
+      }
+  </style>
+  <script>
+      async function sha256 (str) {
+          const buf = await crypto.subtle.digest('SHA-256', new TextEncoder('utf-8').encode(str));
+          return Array.prototype.map.call(new Uint8Array(buf), x => (('00' + x.toString(16)).slice(-2))).join('');
+      }
+      
+      class Playback {
+          constructor (firstElement, outputElement, sampleElement, statsElement) {
+              this.firstElement = firstElement;
+              this.outputElement = outputElement;
+              this.sampleElement = sampleElement;
+              this.statsElement = statsElement;
+              this.firstElement.addEventListener('click', this);
+              this.outputElement.addEventListener('click', this);
+              this.sampleElement.addEventListener('ended', this, false);
+
+              this.offlineAudioContext = new OfflineAudioContext(1, 44100 * 40, 44100);
```

Comment:
I think... this should work in Safari but I don't have a copy to check. As mentioned this won't be an issue soon given they've added a [commit to webkit](https://trac.webkit.org/changeset/265062/webkit) to add support for some of these things.

Reply:
Sorry, it still doesn't work on Safari (14.0.3) :( `renderedBuffer` is undefined (see screenshot). If this can work (Safari supports all relevant APIs) you can use jsbin+browserstack to debug.

<img width="544" alt="undefined" src="https://user-images.githubusercontent.com/985504/112829482-7b67f400-9091-11eb-8264-ab81f4d1a772.png">